Tumor size improvements prior to STU2

masnick opened the pull request:
- [x] Add invariant to enforce providing either `focus` or `specimen` on TumorSize-conforming resources
- [x] Switch measurement method list to SNOMED
- [x] Use fixed code for morphology in Tumor (maybe `367651003` "Malignant neoplasm of primary, secondary, or uncertain origin (morphologic abnormality)")
- [x] Add extension to Tumor to provide a link to a cancer condition, modeled on <http://hl7.org/fhir/R4/extension-condition-related.html>
- [x] Add documentation that elements in Tumor should be invariant during the "lifespan" of the tumor
- [x] Figure out how to handle multifocal tumors
- [x] Use `identifier` to associate Specimen with the corresponding Tumor. To do this I added a TumorSpecimen profile.

mlterryMitre reviewed (commented):
Comments potentially needing further discussion:
- should we consider another component to distinguish multicentric tumors? Currently there's only multifocal.
- there are now 2 different places for which to identify histology/morphology. One at the PrimaryCancerCondition, and now at each tumor. I can see both approaches but this further increases the complexity for mCODE receivers to extract and reconcile histologies for decision support (e.g.: Trials matching, pathways).

masnick commented:
Thanks, @mlterryMitre!

For multicentric vs. multifocal, I wasn't familiar with the difference so I did some research. Per [this article](https://acsjournals.onlinelibrary.wiley.com/doi/full/10.1002/cncr.10331?sid=nlm%3Apubmed):

>  In the currently accepted terminology, multifocality represents the presence of multiple foci of the same tumor, whereas multicentricity suggests different primary tumors in the same breast.

If you agree with this definition, do you think that multicentric would be represented by multiple Tumor-conforming resources vs. a single resource? If so then I don't think we need to have a way to identify a single resource as multicentric.

----

> there are now 2 different places for which to identify histology/morphology

I noticed this too and should have flagged it in the PR description. I completely agree it's not ideal.

If the Tumor referenced the cancer condition then we could rely on that, but I made that connection go the other way (cancer condition → tumor) as there's nowhere in BodyStructure to put the reference to the cancer condition without using an extension.

I was trying to avoid extensions, but I don't love using `CancerCondition.evidence` to connect the tumor and cancer condition anyway. So maybe the better approach is to define an extension on Tumor to reference the cancer condition, and then remove the MS flag on `Tumor.morphology`.

I believe the biggest downside to using an extension for this is that it won't be possible to get all the tumors associated with a given cancer condition unless we define a custom search term (I don't think it's possible to search inside extensions with the standard FHIR search API but I could be wrong about this). Instead you'd have to get all the tumors for the patient, and then filter client-side.

What do you think?

markkramerus commented on input/fsh/SD_TumorSize.fsh:
```diff
@@ -15,7 +15,6 @@ Description:  "Records the dimensions of a tumor"
 * subject 1..1 MS
```
Throughout, ^short should be a name or short description and ^definition should be a description. You don't need to put conformance expectations in the definition, since the `only` statement clearly sets the data type.

mlterryMitre reviewed (commented):
_re: repurposing multifocal and multicentric._ 
- I think they can be represented the same way; it's just that only multifocal is a fixed code. So as long as there's a note that you're basically citing this one source to equate them semantically, then I guess that's okay for ballot.

_re: histology in both tumor and at primary cancer condition._
-  yeh, good point on extensions. For design decisions, I tend to level-set by going back to the driving use case for it. So, if it's finding a specific cohort for trials-matching or pathways, then how would this work? Or if it's to take an mCODE record and translate it for registry reporting, how would that translation work? In both cases, extensions are probably not a good idea. So then, if what's proposed is not ideal, but the most usable of the options, then we probably at least need to document it, and more ideally, socialize the idea with other stakeholders (e.g.: Epic, Trialscope, CDC, etc.).

markkramerus commented:
Considering the tumor size method value set, I think SNOMED has better options:
5880005 | Physical examination procedure (procedure) |
168455000 | Gross pathology (finding) |
104157003 | Light microscopy (procedure) |
363679005 | Imaging (procedure) |
   16310003 | Diagnostic ultrasonography (procedure) |
   113091000 | Magnetic resonance imaging (procedure) |
   77477000 | Computerized axial tomography (procedure) |
   44491008 | Fluoroscopy (procedure) |
I'm not sure how to capture all possible radiology procedures, since all descendants of "imaging" is a big set -- do we want that?

markkramerus commented on input/fsh/SD_TumorSize.fsh:
```diff
@@ -76,6 +84,17 @@ Description:  "Identifies a tumor. Whenever possible, a single resource conformi
 * patient only Reference(CancerPatient)
 
+// Use locationQualifier to identify a multifocal tumor, in addition to modifying body site
+* locationQualifier from http://hl7.org/fhir/ValueSet/bodystructure-relative-location (example)
```
Why not `LocationQualifierVS (required)`? We do that for the location of the PrimaryCancerCondition. Why wouldn't we do that here?

markkramerus commented on input/fsh/SD_TumorSize.fsh:
```diff
@@ -76,6 +84,17 @@ Description:  "Identifies a tumor. Whenever possible, a single resource conformi
+* locationQualifier contains multifocalTumor 0..* MS
+* locationQualifier[multifocalTumor].coding.code = SCT#524008 "Multifocal (qualifier value)"
+* locationQualifier[multifocalTumor] ^short = "Use to identify a multifocal tumor"
+* locationQualifier[multifocalTumor].coding.code ^short = "SNOMED 524008 \"Multifocal (qualifier value)\""
```
LocationQualifier is meant to answer "where", not "what". Multifocal is not an answer to "where".  Multifocal belongs in Morphology.

masnick commented:
### Multifocal tumor approach
 
After checking with a SME, the individual masses in, for example, a multifocal breast cancer would each get their own physical "biopsy clip" to identify them (they wouldn’t necessarily know separate masses were multifocal tumors vs. separate tumors at the time of biopsy). The individual masses would also be measured separately for staging, which would be based on the measurements of the largest mass.
 
Additionally, it seems like it would be possible to get separate measurements for multiple tumors, e.g. from a physical exam or radiology, and then later on these tumors end up being grouped together as a multifocal tumor, e.g. based on pathology.
 
I think our Tumor profile is the FHIR analog of the physical “biopsy clip” that tracks a specific mass — in other words, a mechanism for identifying a specific “abnormal body structure” through time. We therefore only want to include immutable attributes in the Tumor profile that are specific to that BodyStructure -- mutable attributes are Observations about the tumor. “Multifocal” does not seem to be immutable as it could, for example, go from “unknown” to “true” during the course of treatment. And it is not really a property of the "abnormal body structure", but rather is an observation based on the presence and characteristics of multiple masses.
 
Based on this, I do not think the concept of “multifocal” belongs in Tumor.
 
Instead, what if we used a separate Observation resource to identify the observation of “multifocal” with `focus` used to reference the specific Tumor BodyStructures that are part of the multifocal tumor? This Observation could be referenced in `hasMember` of a TNM staging Observation.

masnick commented:
This is ready to re-review. Changes since the last review:

* Remove conformance information from `short` for focus and specimen
* Switch from LOINC to SNOMED for tumor measurement method
* Fix tumor morphology to SCT#367651003
* Switch to using LocationQualifierVS for tumor location qualifier
* Move tumor size-related value sets to the VS_Other.fsh file
* Add extension to link tumor to cancer condition
* Add TumorSpecimen profile, associated examples and documentation
* Make extension for linking tumor to condition generic
* Make effective date MS on TumorSize
* Add note that Tumor elements should be time-invariant
* Add profile for Multifocal Observation, back out slicing for multifocal tumor from cancer condition evidence element
